resources/cf-url-rewriter/cf-url-rewriter.ts

```diff
@@ -19,21 +19,31 @@ export async function handler(event: CloudFrontRequestEvent, context: Context, c
 
   if (request.uri.includes("/latest/")) {
 
-    const indexUri = request.uri.replace(/\/latest\/.*/, '/index.json');
+    const latestPath = request.uri.split("/latest")[1].split("/").slice(0, 4).join("/");
 
-    try {
-      const data: any = await httpsGet('https://' + request.headers.host[0].value + indexUri);
+    const latestIndexUri = request.uri.replace(/\/latest\/.*/, '/index' + latestPath + '/index.json');
+
+    try{
+      const data: any = await httpsGet('https://' + request.headers.host[0].value + latestIndexUri);
 
       if (data && data.latest) {
         callback(null, redirectResponse(request, data.latest));
-      } else {
-        callback(null, errorResponse());
       }
     } catch (e) {
-      console.log(e);
-      callback(null, errorResponse());
-    }
+      try{
+      const oldIndexUri = request.uri.replace(/\/latest\/.*/, '/index.json');
+      const indexData: any = await httpsGet('https://' + request.headers.host[0].value + oldIndexUri);
 
+        if (indexData && indexData.latest) {
+          callback(null, redirectResponse(request, indexData.latest));
+        } else {
+          callback(null, errorResponse());
+        }
+      } catch (e) {
+        console.log(e);
+        callback(null, errorResponse());
+      }
+    }
   } else {
```

Comment:
You can simplify the code by extracting common logic to a sub function. Below is the skeleton.

```

function getLatest(url) {

  try {
    const data: any = await httpsGet(url);

    if (data && data.latest) {
      return data.latest;
    }

    return "";
  } catch (e) {
    console.log(e);
    return "";
  }

}

function handler(() {

  ...
  const latestFromNewUrl = getLatest(newUrl);

  if (latestFromNewUrl) {
    callback(null, redirectResponse(request, latest));
  } else {
    const latestFromOldUrl = getLatest(oldUrl);

    if (latestFromOldUrl) {
      callback(null, redirectResponse(request, latestFromOldUrl));
    } else {
      callback(null, errorResponse());
    }

  }

  ...
}
```